blocks/library/embed/index.js

```diff
@@ -38,6 +65,10 @@ function getEmbedBlockSettings( { title, icon, category = 'embed', transforms, k
 
 		keywords,
 
+		providerName,
```

Comment:
These are not properties of the block API, which we are registering here. It actually makes me think we ought to be warning on non-recognized properties on the registered block instance.

Instead, you could just move `findBlock` into the scope of this function so it has direct access to `patterns`.

Reply:
Where is `providerName` assigned? If it is assigned, should it be assigned as the default for `providerNameSlug`? Might be useful to avoid block unneeded block updates for known block types.

Reply:
`findBlock` needs access to the patterns from all of the embed blocks though, because when the user tries to embed a URL and that URL doesn't match the patterns of the current block, it has to go through all the other embed block settings trying to find one it matches, so I'm not sure moving it into that scope would help? Maybe I'm missing something here though.

Reply:
`providerName` comes from the embed API. The current block titles are based on the provider names, and as there are plugins which do not provide a provider name, I've updated this so that it's not part of the block settings and if we get it from the API we use it, if not we use the title.

Reply:
>`findBlock` needs access to the patterns from all of the embed blocks though, because when the user tries to embed a URL and that URL doesn't match the patterns of the current block, it has to go through all the other embed block settings trying to find one it matches,

Does it? `getEmbedBlockSettinsg` is used to create many individual instances of blocks. Each block should only be responsible for transforming into itself, not other blocks. A block should have no awareness of other blocks existing, even here for embeds.